etl/airflow.md

```diff
@@ -0,0 +1,329 @@
+to schedule DAG runs. We are [actively working on identifying a preferred
+strategy](https://github.com/datamade/la-metro-dashboard/issues/59) for doing
+this while minimizing repeat code and clever architecture.
+
+## Deploying Airflow
+
+In theory, Airflow should be easy to deploy on Heroku. There are a number of
+guides for doing this online (like [this
+one](https://medium.com/@damesavram/running-airflow-on-heroku-ed1d28f8013d)).
+Most use Procfiles instead of the container stack, but our [docs for deploying
+Django on Heroku using the container
+stack](https://github.com/datamade/how-to/blob/master/heroku/deploy-a-django-app.md)
+should be straightforward to adapt for a Flask app like Airflow.
+
+However, our first production Airflow instance, the [LA Metro
+Dashboard](https://github.com/datamade/la-metro-dashboard/), is deployed on AWS
+EC2 via CodeDeploy using DataMade’s [legacy deployment
+framework](https://github.com/datamade/deploy-a-site/). The primary reason we
+had to fall back to our legacy framework is because **we** **needed a stable IP
+address that we could safelist to access an external Solr instance**. Our
+original plan was to expose our Solr instance publicly and enable
+[SSL](https://lucene.apache.org/solr/guide/7_6/enabling-ssl.html) and [basic
+authentication](https://lucene.apache.org/solr/guide/8_1/basic-authentication-plugin.html)
+so that an Airflow instance on Heroku could communicate with it, but we were
+unable to get either feature to work properly before we ran out of budget for
+the task.
+
+Since the IP address of a Heroku app can change without notice, Heroku is a bad
+choice for an Airflow deployment that requires a static IP address. However,
+this is currently the only hurdle to deploying Airflow on Heroku that we know
+of, and we may be able to get around it with more research.
+
+## Tips and Tricks
+
+Since Airflow is a packaged Flask app, debugging it can be challenging. The
+following is an evolving list of tricks that we have identified for debugging
+Airflow.
+
+### Starting a Flask shell
+
+The following command will start a [Flask
+shell](https://flask.palletsprojects.com/en/1.1.x/shell/) in the Airflow app:
+
+``` bash
+# This command assumes your Airflow service in docker-compose.yml is
+named “airflow”
+
+docker-compose run --rm -e FLASK_APP=airflow.www.app airflow flask shell
+```
+
+### Preventing worker refresh when debugging
+
+Airflow needs to periodically refresh its worker process in order to register
+changes to DAGS, and by default it does this every 30 seconds. This can be
+frustrating during debugging because refreshing the worker process will end a
+running process and kick you out of a debugger that you’ve dropped into that
+process.
+
+To prevent reloading when debugging, you can temporarily set the
+[`worker_refresh_interval`](https://github.com/apache/airflow/blob/11eb649d4acdbd3582fb0a77b5f5af3b75e2262c/airflow/config_templates/default_airflow.cfg#L430-L431)
```

Comment:
@jeancochrane I think it would be a good idea to point to this, and the reloading plugin code, in the section on default settings.